src/test/kotlin/org/jmailen/gradle/kotlinter/functional/ExtensionTest.kt

```diff
@@ -119,4 +119,34 @@ internal class ExtensionTest : WithGradleTest.Kotlin() {
             assertEquals(TaskOutcome.SUCCESS, task(":lintKotlinMain")?.outcome)
         }
     }
+
+    @Test
+    fun `extension properties are evaluated only during task execution`() {
+        projectRoot.resolve("build.gradle") {
+            @Language("groovy")
+            val buildScript = """
+                plugins {
+                    id 'kotlin'
+                    id 'org.jmailen.kotlinter'
+                }
+                
+                tasks.whenTaskAdded {
+                    // configure all tasks eagerly
```

Comment:
Here's test that fails on `2.3.0` that quite clearly shows what causes #130 .

`2.2.0` flow:
1. Apply _Kotlinter_ plugin
2. Register hook (but there are no tasks registered so nothing happens)
3. Configure `kotlinter` extension
4. In project's `afterEvaluate` read from extension and configure all tasks
5. Run registered hook on new tasks

`2.3.0` flow:
1. Apply _Kotlinter_ plugin
2. Register tasks, but not configure them
3. Register hook (triggering registered tasks configuration)
4. Configure tasks by reading from an empty extension
5. Configure `kotlinter` extension

Proposed flow reads from `kotlinter` extension in execution phase so the extension will always be filled in.